Pricing/ItemPriceCalculator.php

```diff
@@ -137,16 +164,48 @@ protected function validateNotNull($key, $value)
     public function getItemPrice($item, $currency, $useProductPrice = true)
     {
         $currency = $this->getCurrency($currency);
-        
+
         if ($useProductPrice) {
             $product = $item->getCalcProduct();
-            $price = $this->priceManager->getBulkPriceForCurrency($product, $item->getCalcQuantity(), $currency);
-            $price = $price->getPrice();
-        } else {
-            $price = $item->getPrice();
+
+            //get special price
+            $specialPrice = $this->priceManager->getSpecialPriceForCurrency($product, $currency);
```

Comment:
duplicated code! 
put the logic above into a function and call the function from here